drivers/hmis/app/graphql/mutations/submit_assessment.rb

```diff
@@ -87,7 +87,16 @@ def resolve(input:, assessment_lock_version: nil)
       assessment.assign_attributes(
         user_id: hmis_user.user_id,
         assessment_date: assessment.form_processor.find_assessment_date_from_values,
+        # Use true_user rather than current_user in case of impersonation, matching load_last_user_from_versions.
+        # See comments on CustomAssessments about user, created_by, created_by_hud_user, and updated_by
+        updated_by: true_user,
```

Comment:
Again I probably missed some convo here @martha but I'm not sure if this is the desired behavior. Previously we intentionally set `UserID` based on the "current user" even if they were not the true user, because of some feedback we had gotten about data correction workflows. The true user will always be available in the audit history. It seems odd to me that the UserID and updated_by_user can be different, now.

Reply:
Makes sense. The thought process on my part was probably something like,
1. add `created_by` and `updated_by`. Per the ticket, these refer to the application user and not the HUD user. Use the "true" user because this is being populated from paper trail in the first place, so make it match that behavior going forward.
2. add `created_by_hud_user` based on changing spec, per my comment [here](https://github.com/greenriver/hmis-warehouse/pull/4924/files#r1852242341). But try to make it match the `created_by` and `updated_by`, so invent some kind of concept of a "true hud user". I think this is the leap that you're saying doesn't really make sense.

Let's discuss more when we call later, just trying to articulate the history :)

Possible the solution we will land on could be something like
- remove `created_by` and `updated_by`, continue loading these from paper trail, even though they were the original asks from the ticket they're no longer really needed(?)
- for `created_by_hud_user`, populate this from paper trail, but going forward, populate it similarly to `UserID` (aka don't worry about the "true" user and just use the HUD user.)

Reply:
Ok I see, good point. Since we're currently loading the user from paper trail, we [do resolve](https://github.com/greenriver/hmis-warehouse/blob/75ca057f8f77da8e9754485fdaf8b1751eda28b0/drivers/hmis/app/graphql/sources/paper_trail_versions.rb#L38) the true user in the api right now.